File: mcrs/HIS-MCRS/Areas/ManagementReports/ViewModels/ARPackageAndInvoices.cs
using DataLayer.Model;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Web;

namespace HIS_MCRS.Areas.ManagementReports.ViewModels
{
    public class ARPackageAndInvoices
    {
        [Display(Name = "Category")]
        public int CategoryId { get; set; }
        [Display(Name = "From Date")]
        public DateTime StartDate { get; set; }
        [Display(Name = "To Date")]
        public DateTime EndDate { get; set; }

        public int Deal { get; set; }
        public int ReportOption { get; set; }

        public List<CategoryModel> CategoryList { get; set; }
    }
}

File: mcrs/HIS-MCRS/Scripts/ManagementReports/ARReports/ARPackageAndInvoices.js
function ViewModel(model) {
    self = this;
    self.StartDate = ko.observable(new Date(moment(model.StartDate)));
    self.EndDate = ko.observable(new Date(moment(model.EndDate)));
    self.Deal = ko.observable(model.Deal);
    self.ReportOption = ko.observable(model.ReportOption);
    self.CategoryList = ko.observableArray(model.CategoryList);

    self.SelectedCategory = ko.observable();

    self.init = function () {
        self.CategoryList.unshift({ Id: 0, Name: 'ALL', Code: '' });
    }
    
    self.init();
}
